grayskull/pypi/pypi.py

```diff
@@ -55,9 +55,15 @@ def __init__(
         self._is_strict_cf = is_strict_cf
         self._setup_metadata = None
         self._is_arch = False
+        name = self._canonicalize_name(name)
         super(PyPi, self).__init__(name=name, version=str(version) if version else None)
         self["build"]["script"] = "<{ PYTHON }} -m pip install . -vv"
 
+    @staticmethod
+    def _canonicalize_name(name:str ):
+        """replace all underscores with hyphens in package name"""
+        return name.replace("_","-") 
```

Comment:
While most projects use `-` in the pkg name and `_` in the import name that is not always true. I'm not sure we have enough metadata to figure that out automatically, if so, we should. If not, the implementation here is making an assumption that may, in some rare cases, fail.

TL;DR I'm not against it but I would prefer a metadata based approach first and a fallback to something we all agree on.

Reply:
I agree with @ocefpaf 

we recover the project name from ``setup.py`` and pypi metadata and after it grayskull try to merge both information, right now it discard the name and does not set it, one way to do it would be:

in the method ``_extract_requirements`` inside of the ``PyPi`` class online [781](https://github.com/conda-incubator/grayskull/blob/2e2ab84beb8dbb71a23eb43abcecb5b48eba949a/grayskull/pypi/pypi.py#L781) we have:
```python
        name = metadata["name"]
```
so, one thing which we can do is:
```python
        name = metadata["name"]
        self.set_jinja_var("name", name)
```

that should solve it for now